### ISSUE TITLE
Form Validation - multi field validation functions

### ISSUE THREAD

**Issue by Cougar84** [Semantic-Org/Semantic-UI#4016]
Hi, I've found a lack of multi field validation integrated with form object for example:
my form have a combo box with 3 selection and each option select activate or disable requirements of the other fields ...

Or:
10 fields, validation returns when the sum of the value of that field is 100 ...

Now we've to create a custom functions and use it in the onSuccess function ... and manually add the error to the error list div or display an Alert or something like that... but it's not integrated with form validation.

Could be great having this kind of feature.

**Comment by jlukic** [comment]
You can add custom rules by extending `$.fn.form.settings.rules` with a new function to validate a field.

**Comment by Cougar84** [comment]
Yeah I know I can extend the rules, but the validation is for a single field ... empty, max, min, equals.

If I put a form with 10 fields and I need to validate if the sum of that 10 fields is 100... 
It's not a validation of a single field ... so the "error" class need to be applied on the 10 field (or a container ), not on just one field. Also the prompt must be a single prompt.

Let me take another example ... I have a form with a table. I can add some rows in the table with jquery or knockout js. Well, I need to validate that table to test if there is at least 1 record. The table is not a field so I can't use a field validation or a custom rule.

**Comment by burka** [comment]
@Cougar84 did you find a good workaround?

**Comment by Cougar84** [comment]
@burka no workaround founded. Now I don't use semantic anymore, but I remember that I've made another javascript validation function after form validation.
Ex.
`if (semantic_form_validation() == true) {
  if (my_js_function() == true) {
    // all fine, go ahead
  } else {
    // jquery function to put error message into the error div
  }
}`

**Comment by burka** [comment]
@Cougar84 thanks for your quick answer and the shared hack :-) One more reason to migrate away :-)
